### PR TITLE
🧪 [testing improvement] Add tests for PLC email helpers

### PR DESCRIPTION
**Pull request by OPS-PIvers**
🎯 **What:** The testing gap addressed: Missing tests for getPlcTeammateEmails in plc.ts.
📊 **Coverage:** What scenarios are now tested: Normalization (trim/lowercase), de-duplication, self-exclusion for teammates, and handling of missing/invalid data.
✨ **Result:** The improvement in test coverage: Added 8 new test cases for PLC utility functions.

---
*PR created automatically by Jules for task [7576240836640289688](https://jules.google.com/task/7576240836640289688) started by @OPS-PIvers*

### CODE REVIEW

**Comment by OPS-PIvers** [comment]
@jules

[utils/plc.test.ts](https://github.com/OPS-PIvers/SpartBoard/pull/1513/files/b111510fc532776983a255ebd62c7689f64d8e88#diff-05625697d45f5a1e6a3f29aa5567cd71248297d21894b232f91278e711921109)
Comment on lines +22 to +32
    it('returns normalized and de-duped member emails', () => {
      const emails = getPlcMemberEmails(mockPlc);
      // user-1: 'user-1@example.com'
      // user-2: 'user-2@example.com'
      // user-3: 'user-1@example.com' (duplicate)
      // user-4: 'user-4@example.com' (included because it uses Object.values(plc.memberEmails))
      expect(emails).toHaveLength(3);
      expect(emails).toContain('user-1@example.com');
      expect(emails).toContain('user-2@example.com');
      expect(emails).toContain('user-4@example.com');
    });
@[gemini-code-assist](https://github.com/apps/gemini-code-assist)
gemini-code-assist Bot
[10 minutes ago](https://github.com/OPS-PIvers/SpartBoard/pull/1513#discussion_r3187846857)
 medium

The test confirms that getPlcMemberEmails includes emails from the memberEmails map even if the corresponding UID is not present in memberUids (e.g., user-4). This is inconsistent with getPlcTeammateEmails, which correctly uses memberUids as the source of truth for membership. Since memberUids is defined in types.ts as the authoritative list of current members, getPlcMemberEmails should be refactored to iterate over memberUids instead of Object.values(memberEmails). This prevents stale or orphaned email entries from being included in the result.

Suggested change
    it('returns normalized and de-duped member emails', () => {
      const emails = getPlcMemberEmails(mockPlc);
      // user-1: 'user-1@example.com'
      // user-2: 'user-2@example.com'
      // user-3: 'user-1@example.com' (duplicate)
      // user-4: 'user-4@example.com' (included because it uses Object.values(plc.memberEmails))
      expect(emails).toHaveLength(3);
      expect(emails).toContain('user-1@example.com');
      expect(emails).toContain('user-2@example.com');
      expect(emails).toContain('user-4@example.com');
    });
    it('returns normalized and de-duped member emails', () => {
      const emails = getPlcMemberEmails(mockPlc);
      // user-1: 'user-1@example.com'
      // user-2: 'user-2@example.com'
      // user-3: 'user-1@example.com' (duplicate)
      // user-4: excluded because UID is not in memberUids
      expect(emails).toHaveLength(2);
      expect(emails).toContain('user-1@example.com');
      expect(emails).toContain('user-2@example.com');
      expect(emails).not.toContain('user-4@example.com');
    });
 @OPS-PIvers	Reply...
[utils/plc.test.ts](https://github.com/OPS-PIvers/SpartBoard/pull/1513/files/b111510fc532776983a255ebd62c7689f64d8e88#diff-05625697d45f5a1e6a3f29aa5567cd71248297d21894b232f91278e711921109)
        memberEmails: { 'user-1': 'user-1@example.com' },
      } as Plc;
      expect(getPlcTeammateEmails(singleMemberPlc, 'user-1')).toEqual([]);
    });
@[gemini-code-assist](https://github.com/apps/gemini-code-assist)
gemini-code-assist Bot
[10 minutes ago](https://github.com/OPS-PIvers/SpartBoard/pull/1513#discussion_r3187846874)
 medium

The test suite is missing a case for getPlcTeammateEmails when memberEmails is undefined. Unlike getPlcMemberEmails, the implementation of getPlcTeammateEmails in plc.ts does not safely handle a missing memberEmails object and will throw an error when attempting to access plc.memberEmails[uid]. Adding this test case will highlight the need for a safety check in the utility function.

    });

    it('handles missing memberEmails gracefully', () => {
      const emptyPlc = {
        ...mockPlc,
        memberEmails: undefined,
      } as unknown as Plc;
      expect(getPlcTeammateEmails(emptyPlc, 'user-1')).toEqual([]);
    });

**Comment by OPS-PIvers** [comment]
This PR overlaps with #1517, which also adds tests for `getPlcMemberEmails` and `getPlcTeammateEmails` in `utils/plc.ts`. Both PRs target the same base branch (`dev-paul`) and modify the same test file (`utils/plc.test.ts`), so they will produce a merge conflict.

Please close one of these PRs and consolidate the relevant test cases — including the edge cases identified in the review comments on both PRs — into a single PR.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: utils/plc.test.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      const emails = getPlcMemberEmails(mockPlc);
+      // user-1: 'user-1@example.com'
+      // user-2: 'user-2@example.com'
+      // user-3: 'user-1@example.com' (duplicate)
+      // user-4: 'user-4@example.com' (included because it uses Object.values(plc.memberEmails))
+      expect(emails).toHaveLength(3);
+      expect(emails).toContain('user-1@example.com');
+      expect(emails).toContain('user-2@example.com');
+      expect(emails).toContain('user-4@example.com');
+    });
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply utils/plc.test.ts:33]: Fixed in 3fb8472. `getPlcMemberEmails` now iterates `plc.memberUids` (the authoritative membership list per `types.ts`) instead of `Object.values(plc.memberEmails)`, so stale/orphaned email entries whose UIDs aren't in `memberUids` no longer leak into the result. The test was updated to match: `user-4` (UID not in `memberUids`) is now excluded, and the result length asserts `2` (after de-duping the `user-1` alias). All 9 tests pass.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: utils/plc.test.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    it('returns empty array if selfUid is the only member', () => {
+      const singleMemberPlc = {
+        ...mockPlc,
+        memberUids: ['user-1'],
+        memberEmails: { 'user-1': 'user-1@example.com' },
+      } as Plc;
+      expect(getPlcTeammateEmails(singleMemberPlc, 'user-1')).toEqual([]);
+    });
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply utils/plc.test.ts:79]: Fixed in 3fb8472. Confirmed your prediction — the original `getPlcTeammateEmails` was crashing on `undefined.memberEmails[uid]` when the field was missing. Added `const emails = plc.memberEmails ?? {};` (matching the `getPlcMemberEmails` pattern) plus a `handles missing memberEmails gracefully` test that asserts `[]` is returned.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: utils/plc.test.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    it('excludes the caller (selfUid)', () => {
+      const emails = getPlcTeammateEmails(mockPlc, 'user-1');
+      // memberUids: ['user-1', 'user-2', 'user-3']
+      // filtered: ['user-2', 'user-3']
+      // user-2 -> user-2@example.com
+      // user-3 -> user-1@example.com (Wait, memberEmails[user-3] is ' USER-1@example.com')
+      expect(emails).toHaveLength(2);
+      expect(emails).toContain('user-2@example.com');
+      expect(emails).toContain('user-1@example.com');
+      expect(emails).not.toContain('USER-1@example.com '); // should be normalized
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply utils/plc.test.ts]: Good catch — fixed in 3fb8472. The original test was locking in a bug where the helper would return the caller's own normalized email whenever it was mapped to a different UID alias, which contradicts the documented "skip self before Drive permission grants" contract. Two changes:

1. `getPlcTeammateEmails` now also drops any teammate entry whose normalized email equals the caller's normalized email, not just the entry whose UID equals `selfUid`.
2. The test is reworded to assert the correct contract: when `user-1` is the caller and `user-3`'s email is also `user-1@example.com`, the result is `['user-2@example.com']` and explicitly does **not** contain `user-1@example.com`.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by OPS-PIvers — COMMENTED** [review]
## Automated Code Review

**Date:** 2026-05-06

### Summary
Adds tests for `getPlcMemberEmails`/`getPlcTeammateEmails` and meaningfully tightens both helpers: `getPlcMemberEmails` now iterates `memberUids` (authoritative per `types.ts`) instead of `Object.values(memberEmails)` so orphaned email entries can no longer leak; `getPlcTeammateEmails` now also drops any teammate entry whose email matches the caller's normalized email under a UID alias. The three prior comments (gemini source-of-truth, gemini missing-`memberEmails`, copilot caller-email-alias) are all resolved.

### Code Quality
- The implementation changes are small and well-justified by the test coverage.
- Self-email exclusion logic is documented inline with a clear "why" comment about Drive permission grants.
- Type-suppression patterns match repo conventions.

### Widget Standards
✓ Not applicable.

### Admin Config Standards
✓ Not applicable.

### Firestore & Security
- Behavior tightening: prior to this PR, `getPlcMemberEmails` would return any email present in `memberEmails`, even for UIDs not in `memberUids`. Stale entries would have been granted Drive permissions. The fix prevents that. This is a **defense-in-depth improvement**, not a regression.
- `getPlcTeammateEmails` previously returned the caller's own email when it was aliased under another UID. The new alias-self check is the correct contract per the helper's documented purpose (skip self before Drive grants).

### Regression Risk
- Low. The behavior change is a tightening, not a relaxation. Anyone relying on the old "any email in `memberEmails`" semantics was exposed to the stale-entry leak fixed here.
- **Coordination risk with PR #1517** — same files, overlapping intent. Pick one to merge.

### Missing Tests
- Coverage is comprehensive (9 tests, including the alias-self edge case).

### Overall
**Merge readiness:** Ready
Stronger of the two PLC-tests PRs (#1513 vs #1517). Recommend merging this one and closing #1517.

---
_This review was generated automatically. All findings should be verified by a human before merging._

---
_Generated by [Claude Code](https://claude.ai/code)_
